### PR TITLE
fix(batch): delete >10 messages in legacy sqs processor

### PR DESCRIPTION
**Pull request by whardier**
**No related issue**

## Description of changes:

SQS/Batch - SImple addition to process chunks of records that need removed in 10's.  Counterintuitive and migrating this behaviour to the batch processor scope itself would be too disruptive.

**Checklist**

 

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] Update tests
* [ ] Update docs
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/376ec0a2ac0d2a40e0af5717bef42ff84ca0d1b9/.github/semantic.yml#L2)

## Breaking change checklist

No breaking changes.  Introduces no new exception handling and would fix batch handling of more than 10 records.

**No RFC**:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by whardier** [comment]
Working on updating the tests now that tests are actually looking at the mocked values.. sigh.

**Comment by heitorlessa** [comment]
Hey @whardier could you create an issue first detailing the bug please?

For example, what configuration are you using so that your function receives more than 10 messages per invocation where this is necessary? e.g. large batch size, longer batch window?

If a single function receives a few thousand records with 50% partial failure, this has the side effect of increasing function duration and potentially timing out if one is not careful.

An issue helps us think about potential side effects like these, compromises, what information we need to provide in the docs to help other customers unfamiliar with this scenario, etc.

Thanks a lot!

**Comment by whardier** [comment]
https://github.com/awslabs/aws-lambda-powertools-python/issues/824

**Comment by heitorlessa** [comment]
@mploski before you tackle the redis one, could you help revamp this PR so we can merge it?

Ask. Move to concurrent futures and pass the client to make it thread safe as explained in the review

Background. This is the legacy batch we keep around that we're only deleting 10 messages instead of all failed messages in a batch (could be hundreds).

**Comment by mploski** [comment]
> @mploski before you tackle the redis one, could you help revamp this PR so we can merge it?
> 
> Ask. Move to concurrent futures and pass the client to make it thread safe as explained in the review
> 
> Background. This is the legacy batch we keep around that we're only deleting 10 messages instead of all failed messages in a batch (could be hundreds).

Sure @heitorlessa will look at this at the end of the next week.

**Comment by mploski** [comment]
@whardier @heitorlessa  I will make changes on top of this PR's branch today.

**Comment by mploski** [comment]
@whardier Could you fetch latest changes from upstream into your clone? You're lagging behind a bit :-)

**Comment by mploski** [comment]
@whardier @heitorlessa  Applied suggestions, please review.

**Comment by whardier** [comment]
This looks clean.  I can't test it right away however.  I like it.

File: aws_lambda_powertools/utilities/batch/sqs.py [hunk 1]
```diff
@@ -120,23 +123,39 @@ def _prepare(self):
         self.success_messages.clear()
         self.fail_messages.clear()
 
-    def _clean(self):
+    def _clean(self) -> Optional[List]:
```
Comment (heitorlessa) [inline aws_lambda_powertools/utilities/batch/sqs.py:126]: niiiice one! Thanks

**Review by heitorlessa — APPROVED** [review]
Looks awesome, thank you so much @mploski and also @whardier for kick starting it.

We can now make moves to release several papercuts and the mypy official support now.

Thank you!

**Pull request merged**